From 42f7c42dfe471264e694031d2ac68c4c27152606 Mon Sep 17 00:00:00 2001
From: Nivedita Sarkar <nsarkar@codeaurora.org>
Date: Tue, 6 Aug 2013 10:02:45 -0700
Subject: [PATCH 20/95] ToneGenerator.h: Add C++11 workaround for static const
 float member initialization

C++11 does not permit the initialization of static const float types.
Use a preprocessor macro as a workaround until the entire system can
be upgraded to C++11.

Change-Id: I959e0bcc972ea9a24451ae1e405e7a43066b760a
---
 include/media/ToneGenerator.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/media/ToneGenerator.h b/include/media/ToneGenerator.h
index 898b743..5ef6201 100644
--- a/include/media/ToneGenerator.h
+++ b/include/media/ToneGenerator.h
@@ -208,7 +208,7 @@ private:
     static const unsigned int TONEGEN_MAX_WAVES = 3;     // Maximun number of sine waves in a tone segment
     static const unsigned int TONEGEN_MAX_SEGMENTS = 12;  // Maximun number of segments in a tone descriptor
     static const unsigned int TONEGEN_INF = 0xFFFFFFFF;  // Represents infinite time duration
-    static const float TONEGEN_GAIN = 0.9;  // Default gain passed to  WaveGenerator().
+    #define TONEGEN_GAIN 0.9  // Default gain passed to  WaveGenerator().
 
     // ToneDescriptor class contains all parameters needed to generate a tone:
     //    - The array waveFreq[]:
-- 
2.5.0

